sympy/core/exprtools.py

```diff
@@ -1414,156 +1290,549 @@ def factor_nc(expr):
+            return G, H
+        return None, None
+
+
+    def _factor_once(self, ldegree):
+        """Factor into pairs of polynomials of degree ldegree and self.degree - ldegree, if possible.
+        Coefficients will not be factored or otherwise simplified.
+
+        Return a generator that yields tuples of polynomials.
+
+        Implementation based on Algorithm 2 from https://arxiv.org/abs/1002.3180
+        """
+        from sympy.solvers.polysys import solve_poly_system
+        if self.degree == 0:
+            return
+        if len(self) < 2:
+            return
+
+        if ldegree == 0 or ldegree >= self.degree:
+            return
+            
+        G, H = self._factor_homogeneous(ldegree)
+        if G is None:
+            return
+
+        n = self.degree
+        h = ldegree
+        k = n - ldegree
+        # Future optimization: select the monomials with the least overlap potential
+        Ghat = next(G.monomials())
+        Hhat = next(H.monomials())
+        
+        dummies = []
+        Fhat = self.copy()
+
+        for j in range(1, n + 1):
+            for i in range(1, j):
+                Fhat -= G.copy(h - i) * H.copy(k - j + i)
+
+            if j <= min(h, k):
+                Ghat_L, Ghat_R = Ghat.split(h - j)
+                Hhat_L, Hhat_R = Hhat.split(j)
+                if Ghat_R.term == Hhat_L.term:
+                    # Lines 7-8 of the published algorithm suggest that the term should be
+                    # Ghat_L.term * Hhat_R.term. This doesn't make sense, however - the term is
+                    # supposed to have degree n - j, not n - 2j.
+                    term = _NCMonomial._mul_terms(Ghat.term, Hhat_R.term)
+                    c = 0
+                    if term in Fhat.rep.get(n - j, {}):
+                        c = Fhat.rep[n - j][term].coeff
+                        Fhat.subtract(_NCMonomial(c, term, self.generators, n - j))
+                    dummy = Dummy()
+                    dummies.append(dummy)
+                    # Lines 10-11 of the published algorithm suggest that we should add a monomial
+                    # with term Ghat_L * Hhat_R to both G and H. If this were the case, then in
+                    # step 3 of the example in 3.3 from the paper, we would add -alpha * y to G instead of -alpha.
+                    # This seems to be a second minor mistake in the published algorithm.
+                    #
+                    # There is also a subtlety on handling coefficients not mentioned in the algorithm.
+                    # We have to divide the new H term by the coefficient of Ghat
+                    # (We would have to do the same to the new G term, except for the invariant
+                    # that Hhat never has a coefficient)
+                    G.add(_NCMonomial(dummy, Ghat_L.term, self.generators, h - j))
+                    H.add(_NCMonomial((c - dummy)/Ghat.coeff, Hhat_R.term, self.generators, k - j))
```

Comment:
I would also like to see elaborate more details about this part because this is not part of the original paper, and it's a bit not trivial to figure out how algebra is correct.